[Lim Jia Rui Ryan] ip

ryanlimjr opened the pull request.

Commit:
Add Gradle support

Commit:
-

Commit:
added feature to echo user inputs

Commit:
added class DisplayList to handle storing of user inputs and modified Duke.java to implement DisplayList

Commit:
added class : Task used to handle creation and manipulation of tasks, modified Duke.java to handle additional commands - Done

Commit:
added 3 new subclasses of Task and modified TaskList.java and Duke.java to handle them

Commit:
added some automated text UI testing

Commit:
added exception handling featuree to Duke.java and created DukeExceptions.java to encapsulate Duke exceptions

Commit:
added delete feature in Duke.java, added the exception handling for feature as well

Commit:
implementing the use of enums

Commit:
Modify Event.java Deadline.java TaskList.java Duke.java to handle and format dates

Commit:
add writing of task to file feature in Duke.java.

Commit:
Move exception handling messages to from Duke.java to DukeExceptions.java

Commit:
Modify string formatting of task to file in Task.java

Commit:
Add formating of date & time for event and deadline task.

Commit:
deconflict merging of branch-level-7 and branch-level-8 to master

Commit:
Abstract away methods and variables into the following classes :

1) Ui.java - handle printing of duke messages.
2) Commands.java and its subclasses - handle command directing.
3) parser.java - handle user inputs.
4) storage.java -handle saving of tasks into hardisk

Commit:
Package java files into their respective packages

Commit:
Add JUnit testing for class Task and its subclasses

Commit:
commit previously missed changes

Commit:
Add JavaDoc headers and comments

Commit:
Clean up to abide by coding standards

Commit:
Deconflict merge with branch-Level-9

Commit:
Deconflict merge with branch-A-JavaDoc

Commit:
Merge branch 'branch-A-CodingStandard'

ryanlimjr committed:
Update Task.java

Commit:
Merge branch 'add-gradle-support'
adding graddle support to the project

Commit:
Add gradle support

Commit:
Add Checkstyle for compiance to Java coding standards

Commit:
Fix gradle run error

Commit:
Fix conflict for merge between branch-A-CheckStyle branch and master

Commit:
Update .gitignore

xxzz-tt commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,204 @@
+     */
+
+
+    private void doneHandler(String[] parameters) throws DukeExceptions.NoUndoneTaskException {
```
Should the method come right after the documentation? I noticed the same issue in several other places too.

xxzz-tt commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,204 @@
+
+
+    /**
+     * saves tasks to hard disk
```
Perhaps you can capitalise ’Saves'?

xxzz-tt commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    /**
+     *
```
Perhaps you can add the description for this method?

xxzz-tt commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+        while ((line = br.readLine()) != null) {
+            String[] parameters = line.split("\\|");
+            String taskType = parameters[0];
+            boolean completionStatus = parameters[1].equals("1");
```
Maybe the name of this boolean variable can be changed to isCompleted?

xxzz-tt commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+        FileReader fr = new FileReader(savedTasks); //reads the file
+        BufferedReader br = new BufferedReader(fr); //creates a buffering character input stream
+        String line;
+        while ((line = br.readLine()) != null) {
```
Perhaps writing 'br.readLine() != null' will make the while loop cleaner?

xxzz-tt commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+     * Returns True if all the tasks in the taskList is done.
+     * @return True if all tasks in the taskList are marked done, false otherwise
+     */
+    public boolean allDone() {
```
Perhaps this method can be renamed as areAllDone?

xxzz-tt reviewed (commented):
Very good job overall except for some very minor lapses in the naming of variables.

Commit:
modify Duke.java run to return the response instead of printing it and returning  void

Commit:
implement GUI to the duke bot with correct commands, yet to implement exception handling into the GUI

Commit:
Update DukeExceptions.java to handle errors with the newly implemented GUI

Commit:
Deconflict merge between build.gradle in head and branch-Level-10

Commit:
convert GUI from using JAVAFX to FXML

Commit:
Add assertin statements to Duke.java, Parser.java & TaskList.java

Commit:
improve code quality in terms of shortening long methods, reefactoring complicated expressions

ryanlimjr committed:
Merge pull request #1 from ryanlimjr/branch-A-Assertions

Add assertin statements to Duke.java, Parser.java & TaskList.java

ryanlimjr committed:
Merge pull request #2 from ryanlimjr/branch-A-codeQuality

improve code quality in terms of shortening long methods, reefactorin…

Commit:
Modify Command .java and its resepective subclasses in preparation to abstractt away certain functional details from Duke.java

Commit:
Modify Command .java and its resepective subclasses in preparation to abstractt away certain functional details from Duke.java

Commit:
Modify tasklist.java to handle add each task type individually

Commit:
Create Gradle.yml for continuous integration workflow

Commit:
Modify duke.java and the whole command classes to abastract away the functional processes into the respective commands

Commit:
Merge branch 'modify-Command-1.0'
modify duke.java and each of the commands in the commands package to abstract away certain functionality to the
respective commands
split up tasklist.java addtasks function to handle each type of task specifically
added exception handling funcitonality in the respecitve type of task adding methonds

Commit:
Add minimal baseline for friendlier syntax :

Commit:
Modify files to abide by coding conventions

Commit:
Add 2 features to BCD-Extension
	1) feature to create new aliases for commands
	2) feature to delete the created aliases
Modify DukeException.java to handle exceptions from the 2 new features

Commit:
Modify duke, Command and its subclasses for better interaction with ui and Gui

Commit:
Improve GUI

Commit:
Write JavaDocs where appropriate.

Commit:
Modify respective file to abide by checkstyle.

Commit:
Modify gradle.yml

Commit:
Experiment with markdown

Commit:
Add List feature description to README.md

ryanlimjr committed:
Update README.md

ryanlimjr committed:
Update README.md

ryanlimjr committed:
Update README.md

ryanlimjr committed:
Update README.md

Commit:
Add add task feature description to README.md

Commit:
Add add task feature description to README.md

Commit:
Finish add task feature description to README.md

Commit:
Finish Find by keyword feature and Find by date feature description

Commit:
Finish all feature description in README.MD

Commit:
Modify format for README.md

Commit:
fix event.java and deadline.java to comply with test

Commit:
Update README.md

Commit:
Add Command summary section

ryanlimjr committed:
Update README.md

ryanlimjr committed:
Update README.md

Commit:
update section 4 of README.md

Commit:
update section 4 of README.md
